datasource.go

```diff
@@ -88,6 +100,34 @@ func (ds *sqldatasource) QueryData(ctx context.Context, req *backend.QueryDataRe
 
 }
 
+func (ds *sqldatasource) getDB(q *Query) (*sql.DB, string, error) {
+	// The database connection may vary depending on query arguments
+	// The raw arguments are used as key to store the db connection in memory so they can be reused
+	key := defaultKey
+	db, ok := ds.dbConnections.Load(key)
+	if !ok {
+		return nil, "", fmt.Errorf("unable to get default db connection")
+	}
+	if !ds.EnableMultipleConnections || len(q.ConnectionArgs) == 0 {
+		return db.(*sql.DB), key, nil
+	}
+
+	var err error
+	key = string(q.ConnectionArgs[:])
```

Comment:
```suggestion
	key = string(q.ConnectionArgs)
```

https://play.golang.org/p/uIM21uDBuJ_Y